### PR TITLE
[Numpy] Added operator logaddexp2; added support for zero-size tensor in BinaryBroadcastBackwardUseIn

### PR DESCRIPTION
**Pull request by zoeygxy**
## Description ##
- Added [np.logaddexp2](https://docs.scipy.org/doc/numpy/reference/generated/numpy.logaddexp2.html) and its docs for future rendering. Scalar calculation is supported. 
- Fixed zero-size tensor error in BinaryBroadcastBackwardUseIn (GPU version).

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added [np.logaddexp2](https://docs.scipy.org/doc/numpy/reference/generated/numpy.logaddexp2.html) and its docs for future rendering. Scalar calculation is supported. 
- [x] Fixed zero-size tensor error in BinaryBroadcastBackwardUseIn.

## Comments ##
- The previous error case description: when one operand is non-zero-size (eg. shape of (1, 3)) and the other is zero-size (eg. shape of (2, 0, 3)), the backward calculation on GPU will raise a floating point error. 
- Please merge this PR before merging implementation for np.logaddexp, the similar operator that also depends on the BinaryBroadcastBackwardUseIn fix.

### CODE REVIEW

**Comment by reminisce** [comment]
Please rebase with the upstream numpy branch.

File: src/operator/numpy/np_elemwise_broadcast_op.cu [hunk 1]
```diff
@@ -48,6 +48,14 @@ NNVM_REGISTER_OP(_npi_maximum)
 NNVM_REGISTER_OP(_npi_minimum)
 .set_attr<FCompute>("FCompute<gpu>", BinaryBroadcastCompute<gpu, mshadow_op::minimum>);
 
+NNVM_REGISTER_OP(_npi_logaddexp2)
+.set_attr<FCompute>("FCompute<gpu>", BinaryBroadcastCompute<gpu, mshadow_op::logaddexp2>);
+
+NNVM_REGISTER_OP(_backward_logaddexp2)
+.set_attr<FCompute>("FCompute<gpu>", BinaryBroadcastBackwardUseIn<gpu,  \
```
[the remark this answers is not in the document]
Comment (zoeygxy) [reply src/operator/numpy/np_elemwise_broadcast_op.cu]: Will do, thanks!

File: src/operator/numpy/np_elemwise_broadcast_op.cu [hunk 2]
```diff
@@ -78,5 +86,12 @@ NNVM_REGISTER_OP(_npi_maximum_scalar)
 NNVM_REGISTER_OP(_npi_minimum_scalar)
 .set_attr<FCompute>("FCompute<gpu>", BinaryScalarOp::Compute<gpu, mshadow_op::minimum>);
 
+NNVM_REGISTER_OP(_npi_logaddexp2_scalar)
+.set_attr<FCompute>("FCompute<gpu>", BinaryScalarOp::Compute<gpu, mshadow_op::logaddexp2>);
+
+NNVM_REGISTER_OP(_backward_logaddexp2_scalar)
+.set_attr<FCompute>("FCompute<gpu>", BinaryScalarOp::Backward<gpu,  \
```
[the remark this answers is not in the document]
Comment (zoeygxy) [reply src/operator/numpy/np_elemwise_broadcast_op.cu]: Will do, thanks!

File: src/operator/tensor/elemwise_binary_broadcast_op.h [hunk 3]
```diff
@@ -580,17 +580,19 @@ inline void BinaryBroadcastBackwardUseInImpl(const OpContext& ctx,
...
-  size_t workspace_size_r = ReduceWorkspaceSize<ndim, DType>(
-      s, rgrad.shape_, req[1], ograd.shape_, lhs.shape_, rhs.shape_);
-  size_t workspace_size = std::max(workspace_size_l, workspace_size_r);
-  Tensor<xpu, 1, char> workspace =
-      ctx.requested[0].get_space_typed<xpu, 1, char>(Shape1(workspace_size), s);
-  Reduce<red::sum, ndim, DType, op::mshadow_op::mul, LOP>(s, lgrad, req[0], workspace,
-    ograd, lhs, rhs);
-  Reduce<red::sum, ndim, DType, op::mshadow_op::mul, ROP>(s, rgrad, req[1], workspace,
-    ograd, lhs, rhs);
+  if (inputs[0].shape_.Size() != 0) {
```
[the remark this answers is not in the document]
Comment (zoeygxy) [reply src/operator/tensor/elemwise_binary_broadcast_op.h]: Will do, thanks!

**Comment by zoeygxy** [comment]
This operator and BinaryBroadcastBackwardUseIn fix will be directly moved to master branch. (#15959)

**Pull request closed without merging**